### PR TITLE
Whitespace Cleanup of all SNMP plugins

### PR DESCRIPTION
**Pull request by cycloon**
As discussed in #878: Clean up the mix of tabs and spaces and convert them all to a single style of tabs

### CODE REVIEW

**Comment by cycloon** [comment]
Hm... can't see the conflicts, github is reporting...

**Comment by cycloon** [comment]
Ok, I got this fixed. I had initially based my branch on an old version of the branch stable-2.0

File: plugins/node.d/snmp__df.in [hunk 1]
```diff
@@ -142,19 +142,18 @@ if ($foundpartitions == 0 or defined $partitions{""})
...
-    print "# found $storage with size $partitions{$stor_id->{$storage}}{size}\n" if $Munin::Plugin::SNMP::DEBUG;
+	$partitions{$stor_id->{$storage}}{storage} = $storage;
+	$partitions{$stor_id->{$storage}}{size}    = get_single ($session, $hrStorageSize . $storage);
+	print "# found $storage with size $partitions{$stor_id->{$storage}}{size}\n" if $Munin::Plugin::SNMP::DEBUG;
 
-    if ($partitions{$stor_id->{$storage}}{size} == 0) {
-      delete $stor_id->{$storage} ;
-    }
+	if ($partitions{$stor_id->{$storage}}{size} == 0) {
+	delete $stor_id->{$storage} ;
```
Comment (sumpfralle) [inline plugins/node.d/snmp__df.in]: Here you lost one indentation, or?

File: plugins/node.d/snmp__df.in [hunk 2]
```diff
@@ -209,12 +208,12 @@ if (defined $ARGV[0] and $ARGV[0] eq "config")
 
 foreach my $storage (keys %$stor_id)
 {
-    $partitions{$stor_id->{$storage}}{used}    = get_single ($session, $hrStorageUsed . $storage);
+	$partitions{$stor_id->{$storage}}{used}    = get_single ($session, $hrStorageUsed . $storage);
```
Comment (sumpfralle) [inline plugins/node.d/snmp__df.in]: Maybe remove the duplicate spaces before the equal sign?

File: plugins/node.d/snmp__if_.in [hunk 3]
```diff
@@ -153,104 +153,103 @@ my ($session, $error);
...
+		$session->get_single($ifEntryDescr) ||
+		"Interface $iface";
 
-    if (! ($alias =~ /\d+/) ) {
-	# If there are no numbers in the $alias add the if index
-	$alias .=" (if $iface)";
-    }
+	if (! ($alias =~ /\d+/) ) {
+		# If there are no numbers in the $alias add the if index
+		$alias .=" (if $iface)";
```
Comment (sumpfralle) [inline plugins/node.d/snmp__if_.in]: Maybe add a whitespace after the equal sign?

File: plugins/node.d/snmp__if_multi.in [hunk 4]
```diff
@@ -442,381 +442,380 @@ my $snmpinfo;
...
-    my $alias = $snmpinfoX->{$if}->{ifAlias} || $snmpinfo->{$if}->{ifDescr} || "Interface $if";
+	my $alias = $snmpinfoX->{$if}->{ifAlias} || $snmpinfo->{$if}->{ifDescr} || "Interface $if";
 
-    if (! ($alias =~ /\d+/) ) {
-	# If there are no numbers in the $alias add the if index
-	$alias .=" (if $if)";
-    }
+	if (! ($alias =~ /\d+/) ) {
+		# If there are no numbers in the $alias add the if index
+		$alias .=" (if $if)";
```
Comment (sumpfralle) [inline plugins/node.d/snmp__if_multi.in]: same as above - maybe add a whitespace?

File: plugins/node.d/snmp__netapp_inodeusage_.in [hunk 5]
```diff
@@ -68,56 +68,56 @@ use warnings;
...
 
 if (defined $ARGV[0] and $ARGV[0] eq 'snmpconf') {
-    print "index 1.3.6.1.4.1.789.1.5.4.1.1.\n";
-    foreach (keys %oids) {
-        print "require $oids{$_} [0-9]\n";
-    }
-    exit 0;
+	print "index 1.3.6.1.4.1.789.1.5.4.1.1.\n";
+	foreach (keys %oids) {
+			print "require $oids{$_} [0-9]\n";
```
Comment (sumpfralle) [inline plugins/node.d/snmp__netapp_inodeusage_.in]: one extra level of indentation?

File: plugins/node.d/snmp__rdp_users.in [hunk 6]
```diff
@@ -92,15 +92,14 @@ users.label users
...
+	exit 0;
 }
 
 my($session, $error) = Munin::Plugin::SNMP->session();
 die "Failed to create session: $error" unless $session;
 
 my $process_names = $session->get_entries(-columns => [ $oid ]);
 my $user_count = $process_names ? grep /^rdpclip\.exe$/i, values %$process_names
-                                : 'U';
+																: 'U';
```
Comment (sumpfralle) [inline plugins/node.d/snmp__rdp_users.in]: This is quite a deep level of indentation, or? Do you use two spaces per tab in your editor?

The editor-configurable amount of whitespace makes it impossible to reach universal alignment with tabs. Thus we should probably just increase the level of indentation by one in cases like above?

File: plugins/node.d/snmp__sensors_fsc_bx_fan.in [hunk 7]
```diff
@@ -56,62 +56,62 @@ my $fanDisabled = 2;
...
-    my $fan = $fans->{$key};
-    # Delete sensors with unknown, disabled or unavailable status
-    delete $fans->{$key}
-      if $fan->{status} == $fanUnknown ||
-	$fan->{status} == $fanDisabled ||
-	$fan->{status} == $fanUnavailable;
+	my $fan = $fans->{$key};
+	# Delete sensors with unknown, disabled or unavailable status
+	delete $fans->{$key}
+	if $fan->{status} == $fanUnknown ||
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_bx_fan.in]: Here you lost one level?

File: plugins/node.d/snmp__sensors_fsc_bx_temp.in [hunk 8]
```diff
@@ -71,66 +71,66 @@ my $tempSensorDisabled = 2;
...
-    delete $temps->{$key}
-      if $temp->{status} == $tempSensorUnknown ||
-	$temp->{status} == $tempSensorDisabled ||
-	$temp->{status} == $tempSensorUnavailable;
+	my $temp = $temps->{$key};
+	$temp->{info} = "Blade $temp->{cabinet} sensor $temp->{number}";
+	$temp->{label} = "Blade $temp->{cabinet} $temp->{label}";
+	# Delete sensors with unknown, disabled or unavailable status
+	delete $temps->{$key}
+			if $temp->{status} == $tempSensorUnknown ||
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_bx_temp.in]: one level lower?

File: plugins/node.d/snmp__netapp_diskusage_.in [hunk 9]
```diff
@@ -71,75 +71,73 @@ use warnings;
...
 
 if (defined $ARGV[0] and $ARGV[0] eq 'snmpconf') {
-    print "index 1.3.6.1.4.1.789.1.5.4.1.1.\n";
-    foreach (keys %oids) {
-        print "require $oids{$_} [0-9]\n";
-    }
-    exit 0;
+	print "index 1.3.6.1.4.1.789.1.5.4.1.1.\n";
+	foreach (keys %oids) {
+			print "require $oids{$_} [0-9]\n";
```
Comment (sumpfralle) [inline plugins/node.d/snmp__netapp_diskusage_.in]: one level lower?

File: plugins/node.d/snmp__print_supplies.in [hunk 10]
```diff
@@ -93,70 +93,69 @@ if(!defined ($session)) {
...
+	print "graph_title Supply Level\n";
+	print "graph_args --base 1000 -l 0 --upper-limit 100\n";
+	print "graph_vlabel %\n";
+	print "graph_scale no\n";
+	print "graph_category print\n";
+	print "graph_info This graph represents supplies level\n";
+
+	foreach my $supply (keys(%supplies)) {
+		print "supply$supply.label ".(length($supplies{$supply}{desc})<=$MAXLABEL ?
+				$supplies{$supply}{desc} : substr($supplies{$supply}{desc},0,($MAXLABEL-3))."...");
```
Comment (sumpfralle) [inline plugins/node.d/snmp__print_supplies.in]: one level too high?

File: plugins/node.d/snmp__sensors_fsc_bx_fan.in [hunk 11]
```diff
@@ -56,62 +56,62 @@ my $fanDisabled = 2;
...
+		}
+		print "$id.type GAUGE\n";
 	}
-	print "$id.type GAUGE\n";
-    }
 } else {
-    print get_id($_), '.value ', $fans->{$_}{value}, "\n"
-      for keys %$fans;
+	print get_id($_), '.value ', $fans->{$_}{value}, "\n"
+			for keys %$fans;
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_bx_fan.in]: one extra level?

File: plugins/node.d/snmp__sensors_fsc_bx_temp.in [hunk 12]
```diff
@@ -71,66 +71,66 @@ my $tempSensorDisabled = 2;
...
-    print 'host_name ', $session->hostname(), "\n"
-      unless $session->hostname eq 'localhost';
-      
-    for my $sensor (keys %$temps) {
-	my $id = get_id($sensor);
-	for my $key (qw(label warning critical info)) {
-	    print "$id.$key $temps->{$sensor}->{$key}\n";
+	print 'graph_order ', join(' ', map { get_id($_) } oid_lex_sort keys %$temps), "\n";
+	print 'host_name ', $session->hostname(), "\n"
+			unless $session->hostname eq 'localhost';
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_bx_temp.in]: One level too deep?

File: plugins/node.d/snmp__sensors_fsc_bx_temp.in [hunk 13]
```diff
@@ -71,66 +71,66 @@ my $tempSensorDisabled = 2;
...
+		}
+		print "$id.type GAUGE\n";
 	}
-	print "$id.type GAUGE\n";
-    }
 } else {
-    print get_id($_), '.value ', $temps->{$_}{value}, "\n"
-      for keys %$temps;
+	print get_id($_), '.value ', $temps->{$_}{value}, "\n"
+			for keys %$temps;
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_bx_temp.in]: One level too deep?

File: plugins/node.d/snmp__sensors_fsc_fan.in [hunk 14]
```diff
@@ -49,64 +49,64 @@ my $fanDisabled = 2;
...
-    # Delete sensors with unknown, disabled or unavailable status
-    delete $fans->{$key}
-      if $fan->{status} == $fanUnknown ||
-	$fan->{status} == $fanDisabled ||
-	$fan->{status} == $fanUnavailable;
+	my $fan = $fans->{$key};
+	$fan->{info} = "Cabinet $fan->{cabinet} fan $fan->{number}";
+	# Delete sensors with unknown, disabled or unavailable status
+	delete $fans->{$key}
+			if $fan->{status} == $fanUnknown ||
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_fan.in]: One level too deep?

File: plugins/node.d/snmp__sensors_fsc_fan.in [hunk 15]
```diff
@@ -49,64 +49,64 @@ my $fanDisabled = 2;
...
-    print 'host_name ', $session->hostname(), "\n"
-      unless $session->hostname eq 'localhost';
-      
-    for my $fan (keys %$fans) {
-	my $id = get_id($fan);
-	for my $key (qw(label info)) {
-	    print "$id.$key $fans->{$fan}{$key}\n";
+	print 'graph_order ', join(' ', map { get_id($_) } oid_lex_sort keys %$fans), "\n";
+	print 'host_name ', $session->hostname(), "\n"
+			unless $session->hostname eq 'localhost';
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_fan.in]: One level too deep?

File: plugins/node.d/snmp__sensors_fsc_fan.in [hunk 16]
```diff
@@ -49,64 +49,64 @@ my $fanDisabled = 2;
...
+		}
+		print "$id.type GAUGE\n";
 	}
-	print "$id.type GAUGE\n";
-    }
 } else {
-    print get_id($_), '.value ', $fans->{$_}{value}, "\n"
-      for keys %$fans;
+	print get_id($_), '.value ', $fans->{$_}{value}, "\n"
+			for keys %$fans;
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_fan.in]: One level too deep?

File: plugins/node.d/snmp__sensors_fsc_temp.in [hunk 17]
```diff
@@ -49,66 +49,66 @@ my $tempSensorDisabled = 2;
...
-    delete $temps->{$key}
-      if $temp->{status} == $tempSensorUnknown ||
-	$temp->{status} == $tempSensorDisabled ||
-	$temp->{status} == $tempSensorUnavailable;
+	my $temp = $temps->{$key};
+	$temp->{info} = "Cabinet $temp->{cabinet} sensor $temp->{number}";
+	# Delete sensors with status unknown, disabled or unavailable
+	delete $temps->{$key}
+		if $temp->{status} == $tempSensorUnknown ||
+		$temp->{status} == $tempSensorDisabled ||
```
Comment (sumpfralle) [inline plugins/node.d/snmp__sensors_fsc_temp.in]: one more level?

**Comment by cycloon** [comment]
Thanks for your review. I indeed failed on a few occasions. On others I tried to indent conditional statements two levels deep if they come after the expression. But I will reverse those.

As for additional whitespace, e.g. after the equals sign, I did not look or change for any of those, but I will look into it.

I will make some corrections and commit them as soon as I can.

**Comment by cycloon** [comment]
In general, it would be nice to have some kind of minimal style guide for the plugin contributors. Maybe add this to the wiki and the plugin creation documentation [in the Guide](http://guide.munin-monitoring.org/en/latest/develop/plugins/howto-write-plugins.html#) [or in Best Practices the old Wiki ](http://munin-monitoring.org/wiki/plugin-bcp)

Are there any opinions for munin base we can build upon?

**Comment by sumpfralle** [comment]
> I will make some corrections and commit them as soon as I can.

Ping? :)

**Comment by sumpfralle** [comment]
I think, a bit of whitespace cleanup happened recently all over the plugin directory.
Thus you proposal will not work anymore, I guess.
I would appreciate an updated pull requests for this topic, if you feel like.
Otherwise I will close this one (for housekeeping).

**Comment by sumpfralle** [comment]
Closing due to age and change of code base ...
(feel free to open another proposal - cleanups are always welcome!)

**Pull request closed without merging**